starport_template/lib/widgets/send_money_form.dart

```diff
@@ -23,21 +31,20 @@ class SendMoneyForm extends StatelessWidget {
       child: Column(
         children: [
           CosmosTextField(
-            onChanged: (value) => onAddressChanged(value),
+            onChanged: (value) => widget.onAddressChanged(value),
```

Comment:
you're not updating the `text` field in `onChanged`, thus the CosmosTextField is resetting its value. You have to make sure the `text` passed is always up-to-date with what user has typed in

Reply:
We will have to do that at every place where the `CosmosTextField` is being used. Should we do that everywhere? Keep a `text`, assign it to the `text` property and keep setting it in `onChanged` ?

Reply:
you're right it will be cumbersome and counter intuitive (this is exactly the reason why flutter team introduced the concept of TextController. let me think about a better solution, I'll issue a separate pr for that